README.md

```diff
@@ -0,0 +1,224 @@
+root directory (`/`) in your local `~/mnt/sshfs` directory, you would do the following:
+
+```bash
+mkdir -p ~/mnt/sshfs  # create the ~/mnt/sshfs directory, if it doesn't already exist 
+sshfs john@10.0.0.1:/ ~/mnt/sshfs
+```
+
+### 2. To unmount the remote filesystem:
+
+General syntax:
+```bash
+# For most systems, including Ubuntu
+fusermount -u mountpoint
+
+# For BSD and MacOS, and also works fine on Ubuntu
+umount mountpoint
+```
+
+For the example above:
+```bash
+# For most systems, including Ubuntu
+fusermount -u ~/mnt/sshfs
+
+# For BSD and MacOS, and also works fine on Ubuntu
+umount ~/mnt/sshfs
+```
+
+### 3. Notes:
+1. It is recommended to run SSHFS as a regular user (NOT as root). For this to work, the mount point
+   must be owned by the user. Therefore, mounting into a `mount` or `mnt` directory you create 
+   inside your home directory is a good practice.  
+1. If the username is omitted, SSHFS will use the local username. 
+1. If the directory is omitted, SSHFS will mount the (remote) home directory.
+1. If you need to enter a password, SSHFS will ask for it (actually, it just runs `ssh` which asks
+   for the password if needed).
+
+Also, many `ssh` options can be specified. See the manual pages for _sftp(1)_ (`man 1 sftp`) and
+_ssh_config(5)_ (`man 5 ssh_config`). The remote port number (`-oport=PORT`) is one of the many
+`ssh` options which works with `sshfs`.
+
+
+<a id="build"></a>
+## Build and Install
+
+### 1. General overview
+
+First, download the latest SSHFS release from https://github.com/libfuse/sshfs/releases. On Linux
+and BSD (as opposed to MacOS), you will also need to install [libfuse][libfuse] 3.1.0 or newer. On
+MacOS, you need [OSXFUSE][OSXFUSE] instead. Finally, you need the [Glib][Glib] library with
+development headers, which should be available from your operating system's package manager.
+
+To build and install, we recommend you use [Meson][Meson] (version 0.38 or newer) and
+[Ninja][Ninja]. After extracting the `sshfs` tarball, create a (temporary) build directory and run
+Meson:
+
+```bash
+mkdir build; cd build
+meson ..
+```
+
+Normally, the default build options will work fine. If you nevertheless want to adjust them, you can
+do so with the `meson configure` command:
+
+```bash
+meson configure                # list options 
+meson configure -D strip=true  # set an option
+```
+
+To build, test and install SSHFS, you then use Ninja. Running the tests requires the
+[`py.test`][py.test] Python module. It also requires configuring ssh keys for passwordless  login to
+yourself via _localhost_. See the [detailed notes about ssh key generation and setup](#sshkeygen)
+below.
+
+```bash
+ninja
+python3 -m pytest test/  # (Optional, but recommended) run the tests
+sudo ninja install
+```
+
+### 2. Detailed instructions
+
+_Tested 22 Dec. 2020 on Ubuntu 20.04._
+
+1. Download and `cd` into the source code
+    1. To download the latest source code:
```

Comment:
The people who install from Git should not need these instructions, so I think this can be safely dropped.

Reply:
I'm trying to write with the lowest common denominator in mind, meaning even for the beginner developer who may need this help. With your permission, I'd like to leave these details. Just a handful of years ago I needed these instructions myself, and not having many repos with these instructions slowed my progress into making contributions.

Reply:
That makes sense, thanks for explaining. How about keeping the instructions but moving them into a separate file ("contributor_info.md" or something like that?). That way, the information is available for those who need it without getting in the way of people who don't need it.

Reply:
:+1:, moving how to build from git to `CONTRIBUTING.md` would be good.